### PR TITLE
Clean up ID logic

### PR DESCRIPTION
**Pull request by sarahyurick**
Closes https://github.com/NVIDIA/NeMo-Curator/issues/150

Removes hardcoded `adlr_id` instances in favor of requiring the user to specify an ID themselves.

### CODE REVIEW

File: nemo_curator/utils/script_utils.py [hunk 1]
```diff
@@ -497,11 +497,11 @@ def parse_gpu_dedup_args(description: str) -> argparse.ArgumentParser:
         argumentHelper.parser.add_argument(
             "--input-json-id-field",
             type=str,
-            default="adlr_id",
+            required=True,
             help="The name of the field within each json object of the jsonl "
             "file that assigns a unqiue ID to each document. "
             "Can be created by running the script "
-            "'./prospector/add_id.py' which adds the field 'adlr_id' "
+            "'./prospector/add_id.py' which adds the field "
```
Comment (ayushdg) [inline nemo_curator/utils/script_utils.py:504]: Nit: can we also update the path to add_id in the comment here to the correct dir.

**Review by ayushdg — COMMENTED** [review]
Minor nit, overall lgtm!

Quick note to @ryantwolf and @yyu22, with this change the `id` field is now a required flag when calling the CLI scripts. If CI/other scripts are relying on the default `adlr_id` we should update it to explicitly provide the ID column now.

File: nemo_curator/utils/script_utils.py (hunk 1 above)
Comment (sarahyurick) [reply nemo_curator/utils/script_utils.py]: Updated, thanks!

**Review by ayushdg — APPROVED** [review]
lgtm!

File: tests/test_download.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ def test_incorrect_snapshot_order_news(self):
             start_snapshot = "2021-10"
             urls = get_common_crawl_urls(start_snapshot, end_snapshot, news=True)
 
-    @pytest.mark.flaky(reruns=4)
+    @pytest.mark.flaky(reruns=5)
```
Comment (ryantwolf) [inline tests/test_download.py:127]: Honestly, while you're here, if you don't mind deleting this test for now I think it's for the best. The flaky hasn't been much help. We can mock it later, but it's more hassle than it's worth at this point.

**Review by ryantwolf — APPROVED** [review]
Looks good, we just need to update the tests. Also I left another tangential comment that you can address now if you'd like or later.

File: tests/test_download.py (hunk 2 above)
Comment (sarahyurick) [reply tests/test_download.py:127]: Sure, can remove in a separate PR.

Comment (sarahyurick) [reply tests/test_download.py:127]: Opened https://github.com/NVIDIA/NeMo-Curator/pull/235.

**Pull request merged**